LLMonFHIR/FHIR Interpretation/FHIRInterpretationModule.swift

```diff
@@ -25,14 +26,45 @@ class FHIRInterpretationModule: Module {
     
     
     func configure() {
-        resourceSummary = FHIRResourceSummary(localStorage: localStorage, openAIModel: openAI.model)
-        resourceInterpreter = FHIRResourceInterpreter(localStorage: localStorage, openAIModel: openAI.model)
+        let openAIModelType = UserDefaults.standard.string(forKey: StorageKeys.openAIModel) ?? .gpt4_turbo_preview
```

Comment:
As far as I can see this, a change in the model type at runtime won't update this in the functions. It doesn't seem possible to move the update logic somewhere further down in the logic of the functions. Would probably require a listener on the changes in the user defaults and then regenerate the function? But even then it might not be picked up by the related modules?